SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriverRPC.h

```diff
@@ -0,0 +1,192 @@
+
+		const FRPCMetaData& GetAdditionalData() const { return MetaData; }
+
+		T Data;
+		FName ReceiverName;
+		FRPCMetaData MetaData;
+	};
+
+	WrappedData MakeWrappedData(Worker_EntityId EntityId, T&& Data, uint64 RPCId)
+	{
+		WrappedData wrapper(MoveTemp(Data));
+		wrapper.ReceiverName = RPCName;
+		wrapper.MetaData.Timestamp = FPlatformTime::Cycles64();
+		if (EventTracer)
+		{
+			wrapper.MetaData.ComputeSpanId(RPCName, *EventTracer, SpatialGDK::EntityComponentId(EntityId, ComponentId), RPCId);
+		}
+
+		return wrapper;
+	}
+
+	FName RPCName;
+	Worker_ComponentId ComponentId;
+	SpatialGDK::SpatialEventTracer* EventTracer = nullptr;
+};
+
+class USpatialLatencyTracer;
+
+UCLASS()
+class SPATIALGDK_API USpatialNetDriverRPC : public UObject
+{
+	GENERATED_BODY()
+public:
+	using StandardQueue = SpatialGDK::TWrappedRPCQueue<FSpatialGDKSpanId>;
+
+	USpatialNetDriverRPC();
+	virtual void Init(USpatialNetDriver* InNetDriver, const SpatialGDK::FSubView& InActorAuthSubView,
+					  const SpatialGDK::FSubView& InActorNonAuthSubView);
+
+	void AdvanceView();
+	virtual void ProcessReceivedRPCs();
+	virtual void FlushRPCUpdates();
+	void FlushRPCQueue(StandardQueue& Queue);
+	void FlushRPCQueue(Worker_EntityId, StandardQueue& Queue);
+	TArray<FWorkerComponentData> GetRPCComponentsOnEntityCreation(const Worker_EntityId EntityId);
+
+protected:
+	void MakeRingBufferWithACKSender(FName QueueName, ERPCType RPCType, Worker_ComponentSetId AuthoritySet,
+									 TUniquePtr<SpatialGDK::RPCBufferSender>& SenderPtr,
+									 TUniquePtr<SpatialGDK::TRPCQueue<FRPCPayload, FSpatialGDKSpanId>>& QueuePtr);
+
+	void MakeRingBufferWithACKReceiver(FName QueueName, ERPCType RPCType, Worker_ComponentSetId AuthoritySet,
+									   TUniquePtr<SpatialGDK::TRPCBufferReceiver<FRPCPayload, TimestampAndETWrapper>>& ReceiverPtr);
+
+	virtual void GetRPCComponentsOnEntityCreation(const Worker_EntityId EntityId, TArray<FWorkerComponentData>& OutData);
+
+	struct UpdateToSend
+	{
+		Worker_EntityId EntityId;
+		FWorkerComponentUpdate Update;
+		TArray<FSpatialGDKSpanId> Spans;
+	};
+
+	StandardQueue::SentRPCCallback MakeRPCSentCallback(TArray<UpdateToSend>& OutUpdates);
+	SpatialGDK::RPCCallbacks::DataWritten MakeDataWriteCallback(TArray<FWorkerComponentData>& OutArray) const;
+	SpatialGDK::RPCCallbacks::UpdateWritten MakeUpdateWriteCallback(TArray<UpdateToSend>& OutUpdates) const;
+
+	void FlushUpdates(TArray<UpdateToSend>&);
+	bool CanExtractRPC(Worker_EntityId);
+	bool CanExtractRPCOnServer(Worker_EntityId);
+	bool ApplyRPC(Worker_EntityId, SpatialGDK::ReceivedRPC, const FRPCMetaData& MetaData);
+
+	USpatialNetDriver* NetDriver;
+
+	USpatialLatencyTracer* LatencyTracer;
+	SpatialGDK::SpatialEventTracer* EventTracer;
+
+	TUniquePtr<SpatialGDK::RPCService> RPCService;
+	// TUniquePtr<SpatialGDK::TRPCBufferReceiver<FRPCPayload, TimestampAndETWrapper>> NetMulticastReceiver;
+};
+
+UCLASS()
+class SPATIALGDK_API USpatialNetDriverServerRPC : public USpatialNetDriverRPC
```

Comment:
Could you give an overview of the key differences between the server and client versions/